### PR TITLE
Add node labeler operand

### PR DESCRIPTION
**Pull request by mwalter-habana**
Changes required for node labeler application

### CODE REVIEW

**Comment by fabiendupont** [comment]
@mwalter-habana, could you please provide some explanation about what the node labeler does and why it's needed in the pull request description and commit message?

I know that many commit message in the repository are very light, but for new features it's a good practice to add more details. This can be used to generate the changelog for example.

Other than that, the pull request looks good. I'll let @mresvanis review it as well before merging.

**Comment by mresvanis** [comment]
@mwalter-habana this PR definitely follows the conventions used in this repository, but to dig into it in more depth it would be easier to first resolve any compilation and linting errors.

Depending on what you are using for Golang development, you can probably integrate most of the Golang tools (e.g. [gofmt](https://pkg.go.dev/cmd/gofmt) and [goimports](https://pkg.go.dev/golang.org/x/tools/cmd/goimports)) in your editor/IDE to make it a bit easier. Some examples:

- https://code.visualstudio.com/docs/languages/go
- https://github.com/fatih/vim-go
- https://github.com/dominikh/go-mode.el

In addition, you can use the following Makefile rules to verify the successful compilation, unit tests and linting:

```shell
# build the operator binary
make build

# run unit tests
make test

# run linting
make lint
```

Please feel free to let me know anytime for further information or clarifications.

**Comment by mwalter-habana** [comment]
Created a new commit which passed the following tests with no errors:
make build
make test
make lint

File: config/rbac/node-labeler_role.yaml
Comment (fabiendupont) [inline config/rbac/node-labeler_role.yaml:11]: @mwalter-habana, why does a node labeler use the `privileged` SCC?

Comment (fabiendupont) [reply config/rbac/node-labeler_role.yaml:11]: I guess it's because it needs `hostPath` volume, right?

File: internal/node/labeler/labeler.go
Comment (mresvanis) [inline internal/node/labeler/labeler.go:197]:
```suggestion
	return ctrl.SetControllerReference(cr, ds, r.scheme)
```

File: internal/node/labeler/labeler.go
Comment (mresvanis) [inline internal/node/labeler/labeler.go:126]:
```suggestion
	return r.DeleteNodeLabelerDaemonSet(ctx, cr)
```

File: internal/node/labeler/labeler.go
Comment (mresvanis) [inline internal/node/labeler/labeler.go:73]: We could make this function private, as it's only used internally in this package:

```go
func getNodeLabelerName(cr *hlaiv1alpha1.DeviceConfig) string
```

**Review by mresvanis — COMMENTED** [review]
@mwalter-habana we could add another source of events for the NodeLabeler and NodeMetrics DaemonSets that the controller owns, when [setting up the controller](https://github.com/fabiendupont/habana-ai-operator/pull/22/files#diff-16d0734da7c17eb9f8f798136510c6ad921c1ddb1251f71952d1094a874dafd3L181):

```go
  Owns(&appsv1.DaemonSet{})
```

File: config/hack/CONTAINER_IMAGES
Comment (mresvanis) [inline config/hack/CONTAINER_IMAGES:21]: After our offline communication with @mwalter-habana , this tag should be updated to `0.0.2`.

#### Context

In AWS the naming defers, instead of `Gaudi` it's listed as `Gaudi secured`, which introduced a validation error to the NFD labels generated. The latter included a space, i.e. `habana.ai/hpu.gaudi secured.present=true`, which is invalid.  

```shell
# AWS dl1.24xlarge instance
$ lspci
00:00.0 Host bridge: Intel Corporation 440FX - 82441FX PMC [Natoma]
00:01.0 ISA bridge: Intel Corporation 82371SB PIIX3 ISA [Natoma/Triton II]
00:01.3 Non-VGA unclassified device: Intel Corporation 82371AB/EB/MB PIIX4 ACPI (rev 08)
00:03.0 VGA compatible controller: Amazon.com, Inc. Device 1111
00:04.0 Non-Volatile memory controller: Amazon.com, Inc. NVMe EBS Controller
10:00.0 Ethernet controller: Amazon.com, Inc. Elastic Network Adapter (ENA)
10:1d.0 Processing accelerators: Habana Labs Ltd. HL-2000 AI Training Accelerator [Gaudi secured] (rev 01)
10:1e.0 Processing accelerators: Habana Labs Ltd. HL-2000 AI Training Accelerator [Gaudi secured] (rev 01)
10:1f.0 Non-Volatile memory controller: Amazon.com, Inc. NVMe SSD Controller
20:1d.0 Processing accelerators: Habana Labs Ltd. HL-2000 AI Training Accelerator [Gaudi secured] (rev 01)
20:1e.0 Processing accelerators: Habana Labs Ltd. HL-2000 AI Training Accelerator [Gaudi secured] (rev 01)
20:1f.0 Non-Volatile memory controller: Amazon.com, Inc. NVMe SSD Controller
90:1d.0 Processing accelerators: Habana Labs Ltd. HL-2000 AI Training Accelerator [Gaudi secured] (rev 01)
90:1e.0 Processing accelerators: Habana Labs Ltd. HL-2000 AI Training Accelerator [Gaudi secured] (rev 01)
90:1f.0 Non-Volatile memory controller: Amazon.com, Inc. NVMe SSD Controller
a0:1d.0 Processing accelerators: Habana Labs Ltd. HL-2000 AI Training Accelerator [Gaudi secured] (rev 01)
a0:1e.0 Processing accelerators: Habana Labs Ltd. HL-2000 AI Training Accelerator [Gaudi secured] (rev 01)
a0:1f.0 Non-Volatile memory controller: Amazon.com, Inc. NVMe SSD Controller
```

File: config/rbac/node-labeler_role.yaml
Comment (mwalter-habana) [reply config/rbac/node-labeler_role.yaml:11]: It needs hostPath in order to write to the /etc/kubernetes/node-feature-discovery/features.d/ directory

**Comment by mwalter-habana** [comment]
> @mwalter-habana we could add another source of events for the NodeLabeler and NodeMetrics DaemonSets that the controller owns, when [setting up the controller](https://github.com/fabiendupont/habana-ai-operator/pull/22/files#diff-16d0734da7c17eb9f8f798136510c6ad921c1ddb1251f71952d1094a874dafd3L181):
> 
> ```go
>   Owns(&appsv1.DaemonSet{})
> ```

@mresvanis is that all than needs to be done? 
image

**Comment by mresvanis** [comment]
> > @mwalter-habana we could add another source of events for the NodeLabeler and NodeMetrics DaemonSets that the controller owns, when [setting up the controller](https://github.com/fabiendupont/habana-ai-operator/pull/22/files#diff-16d0734da7c17eb9f8f798136510c6ad921c1ddb1251f71952d1094a874dafd3L181):
> > ```go
> >   Owns(&appsv1.DaemonSet{})
> > ```
> 
> @mresvanis is that all than needs to be done? image

Yes, exactly that.

**Review by mresvanis — APPROVED** [review]

**Pull request merged**
